src/python_routing_v02/troute/routing/fast_reach/mc_reach.pyx

```diff
@@ -902,76 +937,82 @@ cpdef object compute_network_structured_obj(
-              segment_ids.append(segment['id'])
-              buf_view[i, 0] = qlat_array[ segment['id'], int((timestep-1)/qlat_resample)]
-              buf_view[i, 1] = segment['dt']
-              buf_view[i, 2] = segment['dx']
-              buf_view[i, 3] = segment['bw']
-              buf_view[i, 4] = segment['tw']
-              buf_view[i, 5] = segment['twcc']
-              buf_view[i, 6] = segment['n']
-              buf_view[i, 7] = segment['ncc']
-              buf_view[i, 8] = segment['cs']
-              buf_view[i, 9] = segment['s0']
-              buf_view[i, 10] = flowveldepth[segment['id'], timestep-1, 0]
-              buf_view[i, 11] = 0.0 #flowveldepth[segment.id, timestep-1, 1]
-              buf_view[i, 12] = flowveldepth[segment['id'], timestep-1, 2]
-
-            compute_reach_kernel(previous_upstream_flows, upstream_flows,
-                                 len(r), buf_view,
-                                 out_buf,
-                                 assume_short_ts)
-
-            # #a = 120
-            # #weight = math.exp(timestep/-a)
-            # #lastobs = 1
-            # for i, id in enumerate(segment_ids):
-            #     flowveldepth[id, timestep, 0] = out_buf[i, 0]
-            #     #for pos, loid in enumerate(lastobs_ids):
-            #     #    if loid == id:
-            #     #        lasterror = flowveldepth[id, timestep, 0] - lastobs_values[pos]
-            #     #        delta = weight * lasterror
-            #     #        flowveldepth[id, timestep, 0] = flowveldepth[id, timestep, 0] + delta
-            #     flowveldepth[id, timestep, 1] = out_buf[i, 1]
-            #     flowveldepth[id, timestep, 2] = out_buf[i, 2]
-
-            #Copy the output out
-            for i, id in enumerate(segment_ids):
-              flowveldepth[id, timestep, 0] = out_buf[i, 0]
-              flowveldepth[id, timestep, 1] = out_buf[i, 1]
-              flowveldepth[id, timestep, 2] = out_buf[i, 2]
-
-      timestep += 1
+                compute_reach_kernel(previous_upstream_flows, upstream_flows,
+                                        len(r), buf_view,
+                                        out_buf,
+                                        assume_short_ts)
+
+                #Copy the output out
+                for i, id in enumerate(segment_ids):
+                    flowveldepth[id, timestep, 0] = out_buf[i, 0]
+                    flowveldepth[id, timestep, 1] = out_buf[i, 1]
+                    flowveldepth[id, timestep, 2] = out_buf[i, 2]
+           
+        if gages_size:
+
+            for gage_i in range(gages_size):                        
+                usgs_position_i = usgs_positions_list[gage_i]
+                a = 120  # TODO: pull this a value from the config file somehow  
+                for i in range(0,len(usgs_values[0])):
+                    if np.isnan(usgs_values[gage_i, timestep-1]):
+                        da_weight = exp(decay_timesteps[gage_i]/-a)
```

Comment:
This decay coefficient needs to be compared in similar units -- a is 120 minutes, so the `decay_timestep[gage_i]` needs to be converted to a time in minutes (presumably by multiplying by dt).